compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -1565,16 +1566,138 @@ fn linker_with_args<'a, B: ArchiveBuilder<'a>>(
     let cmd = &mut *codegen_results.linker_info.to_linker(base_cmd, &sess, flavor);
     let link_output_kind = link_output_kind(sess, crate_type);
 
-    // NO-OPT-OUT, OBJECT-FILES-MAYBE, CUSTOMIZATION-POINT
+    // ------------------------------ Early order-dependent options -------------------------------
```

Comment:
Nit: Can we reduce the number of `-` or remove them entirely and use some other mechanism (caps?) as a call-out here?

The dashes as used here present multiple problems:

1. It doesn't fit in the github review window, splitting in a fairly weird way:
![20210523_14h33m37s_grim](https://user-images.githubusercontent.com/679122/119258783-bcc0df80-bbba-11eb-9971-8f207c429604.png)
2. In case it becomes necessary to adjust the text for this comment in the future, it is going to be fairly annoying to adjust the number of dashes so that the text remains centered/fits into the required number of columns/etc.